src/PhoneNumberInput/PhoneNumberInput.jsx

```diff
@@ -67,6 +74,14 @@ const PhoneNumberInput = forwardRef((props, ref) => {
     return () => document.removeEventListener('click', onClickOutside)
   }, [])
 
+  useEffect(() => {
+    if (ref) {
+      ref.current.selectionStart = ref.current.selectionEnd = cursorPosition
```

Comment:
What if I want to select something? This will unselect every time the effect runs.
If you need to read and write DOM related values, use `useLayoutEffect`. This way, you make sure all the DOM is initialised.
I would avoid though to change the DOM things in an `useEffect`, depending on a value that changes every time a key is typed.

Reply:
This will not impact the selection flow, because this code runs only when an user modifies the value inside the component. Selection and changing the value, both at the same time, will be impossible.